memberlist.go

```diff
@@ -126,6 +133,18 @@ func newMemberlist(conf *Config) (*Memberlist, error) {
 	return m, nil
 }
 
+func actualBindPort(l net.Listener) (int, error) {
+	_, portStr, err := net.SplitHostPort(l.Addr().String())
```

Comment:
You can just cast the `l.Addr()` to `net.TCPAddr` and get the Port directly! Way simpler than having to convert to string, split and parse.